src/signal.rs

```diff
@@ -134,13 +149,47 @@ pub fn validate_signal_num(num: c_int, for_vcpu: bool) -> errno::Result<c_int> {
     Err(errno::Error::new(EINVAL))
 }
 
-/// Registers `handler` as the signal handler of signum `num`.
+/// Register the signal handler of signum.
+///
+/// # Safety
+///
+/// This is considered unsafe because the given handler will be called
+/// asynchronously, interrupting whatever the thread was doing and therefore
+/// must only do async-signal-safe operations.
 ///
-/// Uses `sigaction` to register the handler.
+/// # Arguments
 ///
-/// This is considered unsafe because the given handler will be called asynchronously, interrupting
-/// whatever the thread was doing and therefore must only do async-signal-safe operations.
-/// flags: SA_SIGINFO or SA_RESTART if wants to restart after signal received.
+/// * `num`: the signal number to be registered.
+/// * `handler`: the signal handler function to register.
+/// * `for_vcpu`: whether `num` is a vcpu signal.
+/// * `flag`: specify the behavior of the signal.
+///    Set SA_SIGINFO or SA_RESTART if wants to restart after signal received.
+///
+/// # Examples
+///
+/// ```
+/// # extern crate libc;
+/// extern crate vmm_sys_util;
```

Comment:
Please comment everything that isn't relevant to the generated example - like `extern crate` definitions, `use`s etc.

Reply:
Right. Fixed.